indigo/tests/test_beautiful_provisions.py

```diff
@@ -1,94 +1,444 @@
+
+    def test_nested_partial_containers(self):
+        # Chapter 1 is mentioned regardless because it's given
+        provision_ids = ['chp_1', 'sec_2']
+        self.beautifier.commenced = True
+        self.assertEqual('Chapter 1 (in part); section 2', self.run_nested(provision_ids))
+        self.beautifier.commenced = False
+        self.assertEqual('Chapter 1 (in part); section 2', self.run_nested(provision_ids))
+
+        # Chapter 1, Part B is mentioned regardless because it's given
+        provision_ids = [
+            'chp_1', 'chp_1__part_A',
+            'sec_1', 'sec_1__subsec_1',
+            'sec_1__subsec_1__list_1__item_a',
+            'sec_1__subsec_1__list_1__item_a__list_1__item_i',
+            'sec_1__subsec_1__list_1__item_a__list_1__item_ii',
+            'sec_1__subsec_1__list_1__item_a__list_1__item_ii__list_1__item_A',
+            'sec_1__subsec_1__list_1__item_a__list_1__item_ii__list_1__item_B',
+            'sec_1__subsec_1__list_1__item_a__list_1__item_iii',
+            'sec_1__subsec_1__list_1__item_aA',
+            'sec_1__subsec_1__list_1__item_b',
+            'sec_1__subsec_1__list_1__item_c',
+            'sec_1__subsec_2', 'sec_1__subsec_3', 'sec_1__subsec_4',
+            'sec_2', 'sec_3',
+            'chp_1__part_B',
+        ]
+        self.beautifier.commenced = True
+        self.assertEqual('Chapter 1 (in part), Part A (section 1–3); Part B (in part)', self.run_nested(provision_ids))
+        self.beautifier.commenced = False
+        self.assertEqual('Chapter 1 (in part), Part A (section 1–3); Part B (in part)', self.run_nested(provision_ids))
+
+        provision_ids = [
+            'chp_1', 'chp_1__part_A',
+            'sec_1', 'sec_1__subsec_1',
+            'sec_1__subsec_1__list_1__item_a',
+            'sec_1__subsec_1__list_1__item_a__list_1__item_i',
+            'sec_1__subsec_1__list_1__item_a__list_1__item_ii',
+            'sec_1__subsec_1__list_1__item_a__list_1__item_ii__list_1__item_A',
+            'sec_1__subsec_1__list_1__item_a__list_1__item_ii__list_1__item_B',
+            'sec_1__subsec_1__list_1__item_a__list_1__item_iii',
+            'sec_1__subsec_1__list_1__item_aA',
+            'sec_1__subsec_1__list_1__item_b',
+            'sec_1__subsec_1__list_1__item_c',
+            'sec_1__subsec_2', 'sec_1__subsec_3', 'sec_1__subsec_4',
+            'sec_2', 'sec_3',
+            'chp_1__part_B', 'sec_4',
+        ]
+        self.beautifier.commenced = True
+        self.assertEqual('Chapter 1 (in part), Part A (section 1–3); Part B (in part); section 4', self.run_nested(provision_ids))
+        self.beautifier.commenced = False
+        self.assertEqual('Chapter 1 (in part), Part A (section 1–3); Part B (in part); section 4', self.run_nested(provision_ids))
+
+        provision_ids = ['chp_1', 'sec_2', 'chp_1__part_B']
+        self.beautifier.commenced = True
+        self.assertEqual('Chapter 1 (in part), section 2; Part B (in part)', self.run_nested(provision_ids))
```

Comment:
Is `Chapter 1 (in part), section 2` the equivalent of `Chapter 1 (in part); Chapter 1, section 2` ? I'm not quite following when the Chapter is included because it's explicitly commenced, and when it's included because it's a parent of the section (ie. to scope the section).

Reply:
I'm starting to think it should be the second option. I had previously thought differently but that was before giving `Chapter 1 (in part)` if it was explicitly marked as commenced.